File: src/main/java/net/kenro/ji/jin/purescript/file/PSFileStubType.java
package net.kenro.ji.jin.purescript.file;

import com.intellij.psi.tree.IStubFileElementType;
import net.kenro.ji.jin.purescript.PSLanguage;

public class PSFileStubType extends IStubFileElementType {
    public static final PSFileStubType INSTANCE = new PSFileStubType();

    private PSFileStubType() {
        super(PSLanguage.INSTANCE);
    }
}

File: src/main/kotlin/org/purescript/file/PSFileStubType.kt
package org.purescript.file

import com.intellij.psi.stubs.PsiFileStub
import com.intellij.psi.tree.IStubFileElementType
import org.purescript.PSLanguage

class PSFileStubType : IStubFileElementType<PsiFileStub<PSFile>?>(PSLanguage.INSTANCE)
val PS_FILE_STUB_TYPE_INSTANCE = PSFileStubType()